openapi.yaml

```diff
@@ -0,0 +1,645 @@
+          schema:
+            $ref: '#/definitions/Errors'
+  /reviews:
+    parameters:
+    - $ref: '#/parameters/authorization'
+    get:
+      summary: List all reviews
+      description: Get a list of all reviews or filter by query parameters
+      tags:
+        - Reviews
+      parameters:
+        - name: reviewer
+          in: query
+          type: string
+          required: false
+          description: Filter by the name of the game reviewer
+        - name: score
+          in: query
+          type: number
+          format: float
+          required: false
+          description: Filter results by the review score
+        - name: reviewDate
+          in: query
+          type: string
+          format: date
+          required: false
+          description: Filter results by posting date
+      responses:
+        '200':
+          description: Successful response
+          schema:
+            $ref: '#/definitions/ReviewResults'
+        '400':
+          description: Bad request
+          schema:
+            $ref: '#/definitions/Errors'
+        '500':
+          description: Internal server error
+          schema:
+            $ref: '#/definitions/Errors'
+    post:
+      summary: Create a review
+      description: Post score and review text to create a review
+      tags:
+        - Reviews
+      parameters:
+      - in: body
+        name: requestBody
+        schema:
+          $ref: '#/definitions/ReviewPostBody'
+      responses:
+        '201':
+          description: Successfully created new review
+          schema:
+            $ref: '#/definitions/ReviewResult'
+        '400':
+          description: Invalid review object
+          schema:
+            $ref: '#/definitions/Errors'
+        '500':
+          description: Internal server error
+          schema:
+            $ref: '#/definitions/Errors'
+  /reviews/{reviewId}:
+    parameters:
+    - $ref: '#/parameters/authorization'
+    - name: reviewId
+      in: path
+      description: Id of review to return
+      required: true
+      type: integer
+      format: int64
+    get:
+      summary: Find review by Id
+      description: Returns all reviews that belong to a particular game
+      tags:
+        - Reviews
+      responses:
+        '200':
+          description: Successful response
+          schema:
+            $ref: '#/definitions/ReviewResult'
+        '404':
+          description: Review not found
+          schema:
+            $ref: '#/definitions/Errors'
+        '500':
+          description: Internal server error
+          schema:
+            $ref: '#/definitions/Errors'
+    patch:
+      summary: Update a review
+      description: Update a review by passing in the reviewId and the parameters to update
+      tags:
+        - Reviews
+      parameters:
+        - name: body
+          in: body
+          schema:
+            $ref: '#/definitions/ReviewResource'
+      responses:
```

Comment:
You can't update the id of a ReviewResource. The review id is passed as a URL parameter.

Reply:
added ReviewPatchBody